transform.go

```diff
@@ -305,3 +322,12 @@ func UnmarshalTransform(v interface{}) (*Transform, error) {
 		return nil, err
 	}
 }
+
+// TransformStep is a unit of operation in a transform script
+type TransformStep struct {
+	Name   string `json:"name"`
+	Path   string `json:"path,omitempty"`
+	Syntax string `json:"syntax"`
+	Type   string `json:"type"`
+	Value  string `json:"value"`
+}
```

Comment:
In lieu of a proper spec we should annotate the purpose of each field in this struct:

```go
type TransformStep struct {
	Name   string `json:"name"`  //  human-readable name for the step, used for display purposes
	Path   string `json:"path,omitempty"` // path to this step if persisted separately on disk
	Syntax string `json:"syntax"`  // execution environment, eg: "starlark", "qri-sql"
	Type   string `json:"type"`  // syntax-specific sub-typing
	Value  string `json:"value"` // input text for transform step. often code, or a query.
}
```